### DISCUSSION TITLE
Profiling of vvdec code

### DISCUSSION THREAD

**Discussion by remul19** [fraunhoferhhi/vvdec#143] [Q&A] [answered]
Hi .. I would like to do a basic profiling of decoder code in linux. Normally we use  **gprof** and while compiling we add ` -pg` option in command line argument, and it creates extra code which is used by gprof command. I went through the build files but could not clearly understand where to add this  ` -pg`  option.  Could anyone please guide on how to do the profiling ?

**Comment by adamjw24** [comment, marked as answer]
As stated in the project description:

```sh
mkdir build
cd build
cmake .. <build options>
cmake --build . # or just `make -j`
```

Just add `-DCMAKE_C_FLAGS=-pg -DCMAKE_CXX_FLAGS=-pg` to your cmake call.

Make sure you are using a clean (empty) directory for a custom CMake build.

**Reply by adamjw24** [reply]
Ah, and probably you would also want to build with `-DCMAKE_BUILD_TYPE=RelWithDebInfo`, so as not to loose symbols for reporting.

**Reply by adamjw24** [reply]
And use `-DCMAKE_VERBOSE_MAKEFILE=1` to double-check if the compilation process does what you wanted it do.

**Reply by remul19** [reply]
Hi ..  I followed the steps as you have suggested.
```
git clone https://github.com/fraunhoferhhi/vvdec
mkdir build
cd build
cmake .. -DCMAKE_C_FLAGS=-pg -DCMAKE_CXX_FLAGS=-pg -DCMAKE_BUILD_TYPE=RelWithDebInfo -DCMAKE_VERBOSE_MAKEFILE=1
cmake --build . 
export PATH=$PATH:/<folder_path>/bin/relwithdebinfo-static
```
`echo $PATH` . Now folder location of vvdecapp is active.
Created an output using command : `vvdecapp -b input.266 -o output.yuv`
Now **gmon.out** is created in **vvdec** folder.
Gave command `gprof vvdecapp gmon.out > analysis.txt` in vvdec folder. Got the error log as `vvdecapp: No such file or directory`.
So copied **gmon.out** into "**bin\relwithdebinfo-static**" folder.
Then navigated to **relwithdebinfo-static** folder: `cd bin\relwithdebinfo-static`
Gave command` gprof vvdecapp gmon.out > analysis.txt.`

Now **analysis.txt** is generated. The input file was a large one and the decodign took nearly 50 seconds  in my Linux machine. But **analysis.txt** is  having 0.00 in most of the columns (except for call count). I was expecting some non-zero values.  A screenshot of **analysis.txt** has been attached for reference. Can anyone please guide me on how to get non-zero values in **analysis.txt** ?

![image](https://github.com/fraunhoferhhi/vvdec/assets/136456486/f160acb0-aa15-40c7-b969-28f008d23aa4)

**Reply by adamjw24** [reply]
Sorry, its up to you to figure out the usage of `gprof`. I never profile under Linux.

**Reply by remul19** [reply]
Hi ..  I tried on another Linux machine, and it worked well with same set of commands.
Screenshot is attached as a reference ...  Thanks 

![image](https://github.com/fraunhoferhhi/vvdec/assets/136456486/3fa66c03-6bfb-48d2-b76d-d315dd190773)

**Reply by 1div0** [reply]
@remul19  Frankly, if you really want to profile the encoder or decoder, use the Intel VTune.
